### PR TITLE
feature/list-sites-subsites-and-doc-libs

### PR DESCRIPTION
**Pull request by jschoewe**
Added actions to list sharepoint site, subsites, and document libraries

### CODE REVIEW

File: actions/doc_lib_list.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/python
+from requests_ntlm import HttpNtlmAuth
+from lib.base_action import SharepointBaseAction
+
+
+class DocLibList(SharepointBaseAction):
```
Comment (nmaludy) [inline actions/doc_lib_list.py:5]: where is the base action?

File: actions/doc_lib_list.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        - domain: Domain for the given username
+        - password: Password to login to sharepoint
+        - site_url: URL of the base Sharepoint site
+        - username: Username to login to sharepoint
+
+        Returns:
+        - List: List of Sharepoint Document Libraries
+        """
+        login_user = domain + "\\" + username
+        user_auth = HttpNtlmAuth(login_user, password)
```
Comment (nmaludy) [inline actions/doc_lib_list.py]: this auth is copy/pasted in every action, would be better to put it into the base

File: requirements-tests.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+mock>=1.3.0,<2.0
+unittest2>=1.1.0,<2.0
+nose>=1.3.7
+pyvmomi>=5.5.0
```
Comment (nmaludy) [inline requirements-tests.txt]: do you need pyvmomi?

File: actions/doc_lib_list.py (hunk 1 above)
Comment (jschoewe) [reply actions/doc_lib_list.py:5]: looks like the gitignore file in cookiecutter-stackstorm includes the lib folder. Adding that now

File: actions/doc_lib_list.py (hunk 2 above)
Comment (jschoewe) [reply actions/doc_lib_list.py]: will do

**Pull request merged**